### ISSUE TITLE
Bug: [no-empty-object-type] should not fire when using interface merging in `declare global`

### ISSUE THREAD

**Issue by jendrikw** [typescript-eslint/typescript-eslint#11468]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play#ts=5.8.2&fileType=.tsx&code=JYWwDg9gTgLgBDAnmApnA3gdWAOwCYQDucAhgM5wASJYYiAIhCNvkQL5wBmUTcARAAsadALQEQfANwAoaXhQBjADYkoaAOZKIAIxJKM0uEbi4YKKJxIK0LAsRQAPM-grVaDJraIHjcNtLYgA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge3oFsbDaOAIwBWiMnSKl0URNGgdokcGAC%2BIFUA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
import type {Window as HappyDomWindow} from "happy-dom";

declare global {
    interface Window extends HappyDomWindow {
    }
}
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/no-empty-object-type": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

No error

### Actual Result

```
An interface declaring no members is equivalent to its supertype. 4:15 - 4:21

> Replace empty interface with a type alias.
```

### Additional Info

AFAIK there's no other way to write this.

The recommended type alias doesn't work because they don't do interface declaration merging, so you get an error about duplicate definitions. But note that this should be flagged:

```
declare global {
    interface Foo extends HappyDomWindow {}
}
```

**Comment by bradzacher** [comment]
https://typescript-eslint.io/rules/no-empty-object-type/#allowinterfaces

You can use `with-single-extends`.

Or If you are choosing to do declaration merging in a way that the rule reports on and in a way that is contravened to the conventions in the rest of your codebase then use a disable comment.
